### PR TITLE
Problem: Ledger does not accept custom derivation path (fix #309)

### LINKED ISSUE #309: Problem: Ledger does not accept custom derivation path
If one use the Devnet with custom derivation path, the address derived is still the default derivation path.

### PR DESCRIPTION
**Pull request by leejw51crypto**
Add address index field

add address index

remove alert

remove console.log

### CODE REVIEW

**Comment by eddycpc** [comment]
Any screenshots of the UI changes ?

**Comment by eddycpc** [comment]
Is the idea of having the custom index supposed to work on ledger only or do we plan to have it working on both ledger and the normal wallets types ?

**Review by Nawaz-Malik143 — DISMISSED** [review]

File: src/pages/create/create.tsx [hunk 1]
```diff
@@ -432,6 +433,24 @@ const FormCreate: React.FC<FormCreateProps> = props => {
...
+          },
+          {
+            pattern: /^0$|^[1-9][0-9]?$|^100$/,
+            message: 'Index between 0 - 100',
+          },
+        ]}
+        hidden={props.isWalletSelectFieldDisable}
+      >
+        <Input placeholder="0" defaultValue="0" />
+      </Form.Item>
```
Comment (eddycpc) [inline src/pages/create/create.tsx:453]: I think this should generally be optional and it should only show if the user is willing to provide a custom derivation index.

cc @crypto-matto @leejw51crypto

**Review by eddycpc — CHANGES_REQUESTED** [review]

**Comment by leejw51crypto** [comment]
image

File: src/pages/create/create.tsx (hunk 1 above)
Comment (crypto-matto) [reply src/pages/create/create.tsx:453]: If so then remove
`initialValue="0"` & `defaultValue="0"`
```
{
            required: true,
            message: 'Please input your address index',
},
```
@leejw51crypto can it auto fill in 0 index at backend when `addressIndex` is empty?

Comment (leejw51crypto) [reply src/pages/create/create.tsx:453]: let's make it visible only for ledger

Comment (eddycpc) [reply src/pages/create/create.tsx:453]: Another question I asked at the top, do we plan to have this working only for Ledger type of wallets?

Do we have the specs requirements to support this feature on normal wallets ?

Comment (leejw51crypto) [reply src/pages/create/create.tsx:453]: only for ledger

Comment (leejw51crypto) [reply src/pages/create/create.tsx:453]: for normal wallets, it can be useful, but can be done as another issue.

Comment (eddycpc) [reply src/pages/create/create.tsx:453]: That's good @leejw51crypto let's have it first working for Ledger as you mentioned. Cool cool!

**Review by linfeng-crypto — APPROVED** [review]

**Comment by linfeng-crypto** [comment]
ZEMU test need fix too

**Comment by leejw51crypto** [comment]
ok, fixing now

**Comment by leejw51crypto** [comment]
ready to merge

**Review by eddycpc — APPROVED** [review]

**Review by eddycpc — APPROVED** [review]

**Review by calvinaco — APPROVED** [review]

**Pull request merged**